pkg/cstorvolumereplica/v1alpha1/cstorvolumereplica.go

```diff
@@ -0,0 +1,61 @@
+package v1alpha1
+
+import (
+	m_k8s_client "github.com/openebs/maya/pkg/client/k8s"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type cvr struct {
+	// name of cstor volume replica
+	name string
+}
+
+type cvrList struct {
+	// option to use to filter
+	// during list execution
+	labelOption string
+
+	// list of cstor volume replicas
+	items []cvr
+}
+
+func (l *cvrList) GetPoolNames() []string {
+	var names []string
+	for _, cvr := range l.items {
+		names = append(names, cvr.name)
+	}
+	return names
+}
+
+type listBuilder struct {
+	list *cvrList
+}
+
+func ListBuilder() *listBuilder {
```

Comment:
I think we should also request for the kubernetes client in this func and return error along with *listBuilder.